core/src/main/java/com/graphhopper/reader/ReaderElement.java

```diff
@@ -41,11 +41,18 @@ protected ReaderElement(long id, int type) {
     }
 
     protected ReaderElement(long id, int type, Map<String, Object> properties) {
+    	validateId(id);
         this.id = id;
         this.type = type;
         this.properties = properties;
     }
 
+    private void validateId(long id) {
+    	if (id < 1) {
+    		throw new InvalidIdException("Invalid Id: " + id + "; Ids must be bigger or equal to 1");
```

Comment:
Can you change this to `OSM Id`, and maybe also include the `type` to make it easier to figure out which element had the negative ID? If you ask me also just inline this method (put the check right into the constructor).

Reply:
adding `type` into the exeption message would only print an Integer. Do you thinks its worth adding a map from e.g. 0 to node and 1 to edge?

Reply:
Can we change `type` from `int` to an enum like `ReaderElementType`? I've been wondering why it is an int anyway.

Reply:
A simple switch should be enough. No need for a map.

(The type should be an enum anyway)

Reply:
something like this?

```
public abstract class ReaderElement {
	public static enum ReaderElementType {
		NODE,
		WAY,
		RELATION,
		FILEHEADER;
	}
    private final long id;
    private final ReaderElementType type;
    private final Map<String, Object> properties;

    protected ReaderElement(long id, ReaderElementType type) {
        this(id, type, new HashMap<>(4));
    }

    protected ReaderElement(long id, ReaderElementType type, Map<String, Object> properties) {
    	if (id < 0) {
    		throw new IllegalArgumentException("Invalid OSM Id: " + id + "; Ids must not be negative");
    	}
        this.id = id;
        this.type = type;
        this.properties = properties;
    }
```

Reply:
Please also add the type to the constructed exception message